src/components/include/connection_handler/connection_handler.h

```diff
@@ -297,7 +297,8 @@ class ConnectionHandler {
   virtual void NotifyServiceStartedResult(
       uint32_t session_key,
       bool result,
-      std::vector<std::string>& rejected_params) = 0;
+      std::vector<std::string>& rejected_params,
+      const std::string reason) = 0;
```

Comment:
@ychernysheva I don't really like this approach to add one more parameter to function with pretty big amount of existing params. Instead of forwarding that string I can propose you to change 2nd parameter from `bool` to `enum` like that:
```
enum class StartServiceResult {
  `kSuccess`, // equals to result == true
  `kGeneralFail`, // result == false and message is empty
  `kFailedSetVideoConfig` // SetVideoCongig response was not successful
  `kTimedOutSetVideoConfig` // SetVideoCongig response was not received
  `kParamsRejected` // Some params were rejected by system
  `kDisallowedHmiLevel` // Streaming rejected for current HMI level
  `kApplicationNotRegistered` // Application was already unregistered
  `kDisallowedAppType` // Application service start rejected due to its HMI type
  `kDisallowedBySettings` // Service start disallowed by settings
  `kDisallowedBySettingsProtectedOnly` // Same as above but for protected services
  `kDisallowedBySettingsUnprotectedOnly` // Same as above but for unprotected services
}
```

Declare that enum somewhere in `protocol_handler.h`
Then declare another function in `protocol_handler_impl.cc` which will accept enum value and generate related string message. This message you can then forward to `SendStartSessionNAck`
Please update all affected functions accordingly and check is that approach simplify things or not.

Reply:
As was noticed, solution with `enums` is not efficient in that case  because sometimes stringified reason could be dynamic and contain information passed from another components.
Was decided to proceed with the initial solution.
@ychernysheva please update your PR accordingly